execute/parallel_test.go

```diff
@@ -0,0 +1,435 @@
+										{Label: "_start", Type: flux.TTime},
+										{Label: "_stop", Type: flux.TTime},
+										{Label: "_time", Type: flux.TTime},
+										{Label: "_value", Type: flux.TFloat},
+										{Label: "_parallel_group", Type: flux.TInt},
+									},
+									Data: [][]interface{}{
+										{execute.Time(5), execute.Time(10), execute.Time(5), 5.0, -1},
+										{execute.Time(5), execute.Time(10), execute.Time(6), 6.0, -1},
+										{execute.Time(5), execute.Time(10), execute.Time(7), 7.0, -1},
+										{execute.Time(5), execute.Time(10), execute.Time(8), 8.0, -1},
+										{execute.Time(5), execute.Time(10), execute.Time(9), 9.0, -1},
+									},
+									ResidesOnPartition: 1,
+								},
+							}),
+						withOutputAttr(plan.ParallelRunKey, plan.ParallelRunAttribute{Factor: 2})),
+					createPhysicalNode("merge", &universe.PartitionMergeProcedureSpec{},
+						withRequiredAttr(plan.ParallelRunKey, plan.ParallelRunAttribute{Factor: 2}),
+						withOutputAttr(plan.ParallelMergeKey, plan.ParallelMergeAttribute{Factor: 2})),
+					createPhysicalNode("filter", &universe.FilterProcedureSpec{
+						Fn: interpreter.ResolvedFunction{
+							Scope: runtime.Prelude(),
+							Fn:    executetest.FunctionExpression(t, "(r) => r._value < 7.5"),
+						},
+					}),
+					createPhysicalNode("yield", executetest.NewYieldProcedureSpec("_result")),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{1, 2},
+					{2, 3},
+				},
+			},
+			want: map[string][]*executetest.Table{
+				"_result": []*executetest.Table{
+					{
+						KeyCols: []string{"_start", "_stop"},
+						ColMeta: []flux.ColMeta{
+							{Label: "_start", Type: flux.TTime},
+							{Label: "_stop", Type: flux.TTime},
+							{Label: "_time", Type: flux.TTime},
+							{Label: "_value", Type: flux.TFloat},
+							{Label: "_parallel_group", Type: flux.TInt},
+						},
+						Data: [][]interface{}{
+							{execute.Time(0), execute.Time(5), execute.Time(0), 1.0, int64(0)},
+							{execute.Time(0), execute.Time(5), execute.Time(1), 2.0, int64(0)},
+							{execute.Time(0), execute.Time(5), execute.Time(2), 3.0, int64(0)},
+							{execute.Time(0), execute.Time(5), execute.Time(3), 4.0, int64(0)},
+							{execute.Time(0), execute.Time(5), execute.Time(4), 5.0, int64(0)},
+						},
+					},
+					{
+						KeyCols: []string{"_start", "_stop"},
+						ColMeta: []flux.ColMeta{
+							{Label: "_start", Type: flux.TTime},
+							{Label: "_stop", Type: flux.TTime},
+							{Label: "_time", Type: flux.TTime},
+							{Label: "_value", Type: flux.TFloat},
+							{Label: "_parallel_group", Type: flux.TInt},
+						},
+						Data: [][]interface{}{
+							{execute.Time(5), execute.Time(10), execute.Time(5), 5.0, int64(1)},
+							{execute.Time(5), execute.Time(10), execute.Time(6), 6.0, int64(1)},
+							{execute.Time(5), execute.Time(10), execute.Time(7), 7.0, int64(1)},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: `parallel-from-filter`,
```

Comment:
_nit_ the first test case was also `parallel-from-filter` and this one just has an extra yield.

Reply:
In the first case only the `from` is parallel. In the second case the `from` and the `filter` run in parallel. I think some comments are in order to clear that up.

Reply:
I added some comments above the tests and also improved the test case names.